src/main/java/spireMapOverhaul/zones/thieveshideout/ThievesHideoutZone.java

```diff
@@ -0,0 +1,137 @@
+import com.megacrit.cardcrawl.monsters.city.Mugger;
+import com.megacrit.cardcrawl.monsters.city.Taskmaster;
+import com.megacrit.cardcrawl.monsters.exordium.Looter;
+import com.megacrit.cardcrawl.monsters.exordium.SlaverRed;
+import com.megacrit.cardcrawl.random.Random;
+import com.megacrit.cardcrawl.rooms.MonsterRoom;
+import com.megacrit.cardcrawl.rooms.MonsterRoomElite;
+import spireMapOverhaul.SpireAnniversary6Mod;
+import spireMapOverhaul.abstracts.AbstractZone;
+import spireMapOverhaul.zoneInterfaces.EncounterModifyingZone;
+import spireMapOverhaul.zones.thieveshideout.events.ThiefKingEvent;
+import spireMapOverhaul.zones.thieveshideout.monsters.BanditLieutenant;
+import spireMapOverhaul.zones.thieveshideout.monsters.ThiefKing;
+import spireMapOverhaul.zones.thieveshideout.monsters.WeakLooter;
+import spireMapOverhaul.zones.thieveshideout.rooms.ForcedEventRoom;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+public class ThievesHideoutZone extends AbstractZone implements EncounterModifyingZone {
+    public static final String ID = "ThievesHideout";
+    private static final String TWO_MUGGERS = SpireAnniversary6Mod.makeID("TWO_MUGGERS");
+    private static final String LOOTER_AND_RED_SLAVER = SpireAnniversary6Mod.makeID("LOOTER_AND_RED_SLAVER");
+    private static final String MUGGER_AND_BLUE_SLAVER = SpireAnniversary6Mod.makeID("MUGGER_AND_BLUE_SLAVER");
+    private static final String THREE_LOOTERS_WEAK = SpireAnniversary6Mod.makeID("THREE_LOOTERS_WEAK");
+    public static final String BANDIT_LIEUTENANT_AND_TASKMASTER = SpireAnniversary6Mod.makeID("BANDIT_LIEUTENANT_AND_TASKMASTER");
+    public static final String THIEF_KING = SpireAnniversary6Mod.makeID("THIEF_KING");
+
+    public ThievesHideoutZone() {
+        super(ID, Icons.MONSTER);
+        this.width = 2;
+        this.height = 5;
+    }
+
+    @Override
+    public AbstractZone copy() {
+        return new ThievesHideoutZone();
+    }
+
+    @Override
+    public Color getColor() {
+        return new Color(0.2f,0.6f, 0.7f, 1f);
+    }
+
+    @Override
+    public boolean canSpawn() {
+        return this.isAct(2);
+    }
+
+    @Override
+    protected boolean canIncludeEarlyRows() {
+        //Since this zone always has a dangerous event node (around super-elite difficulty), we don't want it to show up in the rows of the act that never have elites
+        //It also has normal fights that are tuned as hard pool fights, so that's another reason to prevent early spawns
+        return false;
+    }
+
+    @Override
+    public void manualRoomPlacement(Random rng) {
+        //We put the special node with the Thief King fight in the last row of the zone (that isn't a protected row)
+        List<MapRoomNode> possibleNodes = new ArrayList<>();
+        for (MapRoomNode node : this.nodes) {
+            if (node.getRoom() == null && !this.isProtectedRow(node.y)) {
```

Comment:
Being able to write this check is why I made `AbstractZone.isProtectedRow` protected instead of private.